include/search_index.h

```diff
@@ -0,0 +1,13 @@
+/** .
+ * @file search_index.h
+ * for doxy
+ *
+ *
+ */
+#ifndef SEARCH_INDEX_H
+#define SEARCH_INDEX_H
+
+
+int search_index(int num);
```

Comment:
Bezüglich dem Search Index kann ich euch gerne am Montag beim Meeting genauere Infos geben :)
Der hintergrund kommt vom dem dass die Arrays mit 8_16_64 Zahlen in einem Doppel Pointer gespeichert sind dass heißt z.b. das array mit 64 ist z.b. folgendermaßen zugreifbar:

```*(input[index] + i)```

Wobei man mit index zwischen den Arrays wechselt und mit " i " durch die elemente durchgehen kann. Ähnlich wie bei einem Hashmap wird hier der Index berechnet wo sich das array befindet. Mittels der Funktion search_index(). Der hintergrund kommt von dem dass wir beim Leistungsvergleich auch ein random input haben. Ich wollte aber eine funktion die so dynamisch und flexibel wie möglich ist, und damit dieses **input mehrfach anwendbar ist bin ich auf diese Lösungsgekommen. Das ermöglicht uns den input für 1.1 sowie 1.2 zu verwenden. Es muss lediglich nur neu aufgefüllt werden. Umso weniger mallocs umso weniger leaks.